app/handlers/oauth_callback.rb

```diff
@@ -27,37 +27,39 @@ def authorized?
     true
   end
 
-  def setup
-    @logged_in_user = options[:logged_in_user]
-
-    @oauth_provider = oauth_data.provider
-    @oauth_uid = oauth_data.uid.to_s
-    outputs.email = oauth_data.email
-  end
-
   def handle
-    if @logged_in_user
-      handle_while_logged_in(@logged_in_user.id)
-    elsif mismatched_authentication?
+    # Possible errors that can occur - so we'll check those first
+    if mismatched_authentication?
       fatal_error(code: :mismatched_authentication)
-    elsif (outputs.authentication = Authentication.find_by(provider: @oauth_provider, uid: @oauth_uid))
-      # User found with the given authentication. We will log them in.
-      outputs.authentication.user
-    elsif (existing_user = user_most_recently_used(users_matching_oauth_data))
-      # No user found with the given authentication, but a user *was* found with the given email address.
-      # We will add the authentication to their existing account and then log them in.
-      outputs.authentication = Authentication.find_or_initialize_by(provider: @oauth_provider, uid: @oauth_uid)
-      run(TransferAuthentications, outputs.authentication, existing_user)
+    # The user is trying to sign up but they came from the login form, so redirect them to the sign up form
     elsif request.env['omniauth.origin']&.to_sym == :login_form
-      # The user is trying to sign up but they came from the login form, so redirect them to the sign up form
       fatal_error(code: :should_redirect_to_signup)
-    else # sign up new student, then we will log them in.
-      user = create_student_user_instance
-      run(TransferOmniauthData, oauth_data, user)
-      outputs.authentication = create_authentication(user, @oauth_provider)
     end
 
-    outputs.user = outputs.authentication.user
+    @authentication = Authentication.find_or_initialize_by(provider: @oauth_data.provider, uid: @oauth_data.uid)
+    user = User.find_by(id: @authentication.user_id)
+    # User found with the given authentication. We will log them in.
+    # byebug
+    if user.present?
+      outputs.user = user
+      handle_while_logged_in(@authentication.user)
```

Comment:
This method should be called if this handler is used by a user that is already logged in (session exists), so that they can add a new social auth or something, but that's not what's happening here.

Reply:
I see - seems like just removing this and letting the social_auth_controller handle the sign_in while outputting the user and authentication is the thing to do here.. let me know if that's not right.